### DISCUSSION TITLE
Proper dependency injection (DI)

### DISCUSSION THREAD

**Discussion by Knoxvillekm** [arkivanov/Decompose#239] [Q&A] [answered]
How to use it, any sample plz.

**Comment by arkivanov** [comment]
Hello. You can just pass things via constructors, see [example](https://github.com/arkivanov/Decompose/blob/e720a0d2230ca530441bbed72efb97af5d8c06c2/sample/shared/shared/src/commonMain/kotlin/com/arkivanov/sample/shared/counters/CountersComponent.kt#L47).

**Comment by plusmobileapps** [comment, marked as answer]
Like @arkivanov mentioned, you just need to pass things via the constructor so the necessity of a DI not required. If you were wanting to use some DI framework, it's worth mentioning that `ComponentContext` must be injected at run time because children are created with a new context from the router. So if you were to use something like kotlin-inject or kodein then you would have to use assisted injection. When I was messing around with decompose in an Android only app, I managed to get it working with Hilt if you wanted to check it out [here](https://github.com/plusmobileapps/decompose-android/compare/main...hilt#diff-2864c73fd2753928b76530fcfb77a11c050b0cfe3f66265d4bd98881c2e7f3e5R14). Otherwise I have another [sample app that just creates is own DI](https://github.com/plusmobileapps/rick-and-morty-kmp/blob/main/sample/shared/src/commonMain/kotlin/com/plusmobileapps/rickandmorty/di/DI.kt) using the features of the kotlin language.
